source/base/mpi.cc

```diff
@@ -622,13 +622,49 @@ namespace Utilities
           return;
         }
 
+      /*
+       * Create a custom MPI datatype handle describing the memory layout
+       * of the MinMaxAvg struct and a custom MPI op handle for our
+       * max_reduce function.
+       *
+       * This setup will leak some minor allocated memory (in the range of
+       * a few bytes) at the end of the program run because we never call
+       * MPI_Type_free(&type) and MPI_Op_free(&op) to explicitly free up
+       * memory. This is a cosmetic issue but some address sanitizers will
+       * complain about this (such as clang's -fsanitize=address).
+       */
+      static MPI_Datatype type;
+      static MPI_Op       op;
```

Comment:
You can do this easier than using `std::call_once` if you do 
```suggestion
      static MPI_Op       op = [=]() {...} ();
```
In other words, you just provide an initializer for the static variable.

Reply:
I thought about this as well.

I opted for the `std::call_once` variant because I think it is mildly cleaner. (We cannot call to `MPI_Type_create_struct`, `MPI_Type_commit` and `MPI_Op_create` outside of an `MPI_InitFinalize` so all of the static initializer would need a separate `if(job_supports_mpi())`.)

Reply:
No, that's what local `static` variables do: They are initialized upon first go-through of the function. They do exactly what you are doing with the `call_once`.

Reply:
I will revise.

Reply:
`std::call_once` is now gone.

Reply:
> They are initialized upon first go-through of the function.

Yes, and the compiler even generates code to make it thread-safe.